### ISSUE TITLE
We are getting where: nil in has_many

### ISSUE THREAD

**Issue by olafura** [elixir-ecto/ecto#2902]
In Ecto 3.0.6:

I think the `def combine_assoc_query(%{where: nil}, queryable), do: queryable` was possible accidentally removed in:
https://github.com/elixir-ecto/ecto/commit/d707901888c76057435f1399b2e625105b834238

Because it blows up with: 
```
 ** (Protocol.UndefinedError) protocol Enumerable not implemented for nil ...
        (elixir) /home/ubuntu/bob/tmp/b6b304296ade3b61cddf7cdf277daacf/elixir/lib/elixir/lib/enum.ex:1: Enumerable.impl_for!/1
        (elixir) /home/ubuntu/bob/tmp/b6b304296ade3b61cddf7cdf277daacf/elixir/lib/elixir/lib/enum.ex:141: Enumerable.reduce/3
        (elixir) lib/enum.ex:1919: Enum.reduce/3
        (ecto) lib/ecto/association.ex:288: Ecto.Association.expand_where/5
        (ecto) lib/ecto/association.ex:282: Ecto.Association.combine_assoc_where/2
```

This is the schema:
```
has_many :claims, Claim, foreign_key: :claimant_id
```

This works in 3.0.4

**Comment by josevalim** [comment]
Where defaults to an empty list now though: https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/association.ex#L572

Can you please remove `_build` and let us know if it works?

**Comment by olafura** [comment]
That works, thanks ;)

**Comment by josevalim** [comment]
Btw, the need to clean _build should be fixed in Elixir v1.8.0. :)

**Comment by olafura** [comment]
@josevalim that might make it easier to convince my team to update ;)
